### PR TITLE
Download submodules and images on demand

### PR DESCRIPTION
**Pull request by wom-bat**
instead of having to have a separate target for getting them.

### CODE REVIEW

File: examples/kitty/kitty.mk [hunk 1]
```diff
@@ -164,16 +163,42 @@ nfs.elf: nfs/nfs.a libnfs/lib/libnfs.a musllibc/lib/libc.a
...
+	git submodule update --init $(LIONSOS)/dep/libvmm
+
+${LIBNFS}/nfs:
+	git submodule update --init $(LIONSOS)/dep/libnfs
+
+$(LIONSOS)/dep/micropython/py/mkenv.mk ${LIONSOS}/dep/micropython/mpy-cross:	
+	git submodule update --init $(LIONSOS)/dep/micropython
+	cd ${LIONSOS}/dep/micropython && git submodule update --init lib/micropython-lib
+
+	git submodule update --init $(LIONSOS)/dep/musllibc
```
Comment (Ivan-Velickovic) [inline examples/kitty/kitty.mk:199]: This one should be for libnfs.

**Comment by Ivan-Velickovic** [comment]
I think this should work, I made the submodules a separate target because I wanted people to develop/change the submodules and be able to test the Kitty example without having all the submodules reset by the Makefile. I think these changes should work in that case though.

Can you make the same submodules changes for the web server as well?

**Comment by wom-bat** [comment]
> I think this should work, I made the submodules a separate target because I wanted people to develop/change the submodules and be able to test the Kitty example without having all the submodules reset by the Makefile. I think these changes should work in that case though.
> 
> Can you make the same submodules changes for the web server as well?

Want to get this one working on the CI first.

**Comment by wom-bat** [comment]
> Want to get this one working on the CI first.

Seems to be a race with -j2 --- works with -j1.  I think I've fixed it, testing now ...

File: examples/kitty/kitty.mk [hunk 2]
```diff
@@ -164,16 +160,45 @@ nfs.elf: nfs/nfs.a libnfs/lib/libnfs.a musllibc/lib/libc.a
...
+${LINUX}:
+	curl -L ${LIONSOS_DOWNLOADS}/$(KITTY_GRAPHICS_VM_LINUX) -o $@
+
+${INITRD}:
+	curl -L ${LIONSOS_DOWNLOADS}/$(KITTY_GRAPHICS_VM_ROOTFS) -o $@
+
+# Because we include the *.mk files they are an implicit dependency
+# for the build.  These rules instantiate the submodules that
+# include the makefiles.
+${LIBVMM_DIR}/vmm.mk:
```
Comment (Ivan-Velickovic) [inline examples/kitty/kitty.mk:186]: These check for the *existence* of these files not if the files have changed right? E.g if you pulled a new version of libvmm that changed `vmm.mk` would it update the submodule?

**Comment by wom-bat** [comment]
> > Can you make the same submodules changes for the web server as well?
Done

**Pull request closed without merging**